awx/main/analytics/collectors.py

```diff
@@ -0,0 +1,218 @@
+    return counts
+
+@register('instance_info')   # maybe integrate this with job_instance_counts?
+def instance_info(since):
+    info = {}
+    instances = models.Instance.objects.values_list('hostname').annotate().values('uuid', 'version', 'capacity', 'cpu', 'memory', 'managed_by_policy', 'hostname')
+    for instance in instances:
+        info = {'uuid': instance['uuid'],
+                'version': instance['version'],
+                'capacity': instance['capacity'],
+                'cpu': instance['cpu'],
+                'memory': instance['memory'],
+                'managed_by_policy': instance['managed_by_policy'],
+                }
+    return info
+
+
+@register('job_counts')
+def job_counts(since):
+    counts = {}
+    counts['total_jobs'] = models.UnifiedJob.objects.all().count()
+    counts['status'] = dict(models.UnifiedJob.objects.values_list('status').annotate(Count('status')))
+    counts['launch_type'] = dict(models.UnifiedJob.objects.values_list('launch_type').annotate(Count('launch_type')))
+    
+    return counts
+    
+    
+@register('job_instance_counts')
+def job_instance_counts(since):         #TODO: all of these are going to need to be restrained to the last 24 hours/INSIGHTS_SCHEDULE
+    counts = {}
+
+    job_types = models.UnifiedJob.objects.values_list(
+        'execution_node', 'launch_type').annotate(job_launch_type=Count('launch_type'))
+    for job in job_types:
+        counts.setdefault(job[0], {})[job[1]] = job[2]
+        
+    job_statuses = models.UnifiedJob.objects.values_list(
+        'execution_node', 'status').annotate(job_status=Count('status'))
+    for job in job_statuses:
+        counts.setdefault(job[0], {})[job[1]] = job[2]
+        
+# See if we can group by status and type inside the node grouping.  ^^
+
+    # 'jobs_running': models.UnifiedJob.objects.filter(execution_node=instance.hostname, status__in=('running', 'waiting',)).count(), # jobs in running & waiting state
+    # 'jobs_total': models.UnifiedJob.objects.filter(execution_node=instance.hostname).count(),
+    # ^^ try to get jobs_running and jobs_total per instance.  and uuid if easily possible  
+
+    return counts
+
+
+# Copies Job Events from db to a .csv to be shipped
+def copy_events_table(path='/tmp/'):
+    events_file = os.path.join(path, 'events_table.csv')
+    try:
+        write_data = open(events_file2, 'w')
+        with connection.cursor() as cursor:
+            cursor.copy_expert('''COPY (SELECT main_jobevent.id, 
+                                      main_jobevent.created, 
+                                      main_jobevent.modified, 
+                                      main_jobevent.event, 
+                                      main_jobevent.failed, 
+                                      main_jobevent.changed, 
+                                      main_jobevent.uuid, 
+                                      main_jobevent.playbook, 
```

Comment:
@matburt @wenottingham do we need this?  You could _probably_ argue that this **could** constitute personally identifiable information.

Reply:
The usage case is:
- find long running / often failing (outlier) tasks
- sum usage of modules/roles
- track deprecation warnings

We should collect the minimum set of data that allows for that.

Reply:
The issue I could see that would require some sort of playbook/play/role/task gathering would be how you'd refer back to the outlier once you find it.

Reply:
I understand, I just want to be clear that while limiting what we collect here _could_ hamper our ability to report, this data _could_ also represent PII.  So we have a judgement call to make.

Reply:
Just to be clear: if there are fields here that are not relevant to performing those calculations, etc - we should update this code to not collect them. I am not specifically reviewing the field list.

Reply:
@wenottingham @rooftopcellist I've thought about this, and I think we _shouldn't_ capture `main_jobevent.task` and `main_jobevent.play`, because they're very likely to contain free-form text and potentially PII.

I think `main_jobevent.playbook` is probably pretty important, though, so we can communicate with users *which* specific playbooks might be using Ansible poorly.

Reply:
Ok, I've removed those fields, as well as:
```
                              main_jobevent.counter, 
                              main_jobevent.start_line, 
                              main_jobevent.end_line, 
                              main_jobevent.parent_id, 
                              main_jobevent.parent_uuid ```

Reply:
I left a comment here:

https://github.com/ansible/awx/pull/3284/files#r267120036

I think if we exclude this data then we'll lose a lot of dimensionality. We *can* convert this information to something else on the backend... hash it or something, but we will lose the ability to drill into any sort of detail when analyzing events across playbook and job runs.

Reply:
As useful as playbook data is, more and more detail is pushed down to roles (and thus plays). Having play/task information lets us correlate roles and plays even *between* job templates and being able to detect when plays change position under a playbook and track performance and characteristics in a larger scope.